### DISCUSSION TITLE
xDrip-plus with Bubble seems to need OOP2 with Libre 2 (UK) sensors

### DISCUSSION THREAD

**Discussion by nigelkeirross** [NightscoutFoundation/xDrip#2568] [Q&A]
I have used xDrip-plus with a bridge for Libre 1 for about 4 years - from LimiTTer, to mm2, and now Bubble & Bubble Mini. Yesterday & started my first Libre 2 (UK) sensor. Seeking a gradual transition,  I decided to go for the more gradual transition - using a bridge - rather than move to the bridgeless (minimallooper of juggluco) direct connection.
I used my Bubble, upgraded to the latest FW (3.8), which I believe is more than sufficient to decode the Libre 2 data.
When I tried to get readings, none were appearing. I looked at the logs, and was surprised to see:
LibreOOPAlgorithm - OOP is not alive, sending data but no response.
I had not changed any settings from Libre 1 usage.
I am using the xDrip-plus from 22-08-03.
Why does it require OOP2 to be installed?
In seetings I do have both Advanced Settings for Libre 2 (Show Raw Vaules & Show sensors info) turned ON. But in misc settings, OOP Libre algorithm is unchecked, and OOP2 algorith calibration is set to "Calibrate based on raw". I believe these to tbe the defaults.
I did install OOP2, and I did get readings, but it appears that the data being returned from OOP2 is the same as the data being sent (see 2nd screenshot), and other documentation states OOP2 should not be necessary with Bubble & Libre 2.
I tried with Bubble Mini too, and it behaved same as Bubble.
Finally, since this xDrip-plus & OOP2 combination for Libre2, battery usage of xDrip has risen significantly - from negligible, to being, at 15%, my highest consumer. Is this to be expected
![dataBeforeAndAfterOOP](https://user-images.githubusercontent.com/21008320/208700239-c10bb14e-a58e-4331-b2e8-b0281dc64afa.png)

**Comment by tzachi-dar** [comment]
Why are you still using bubble? you can use libre2 without.

**Reply by kleenkanteen** [reply]
How? Do I just use xDrip+ and Diabox, and send the data from Diabox to xDrip+? Any more info? I followed one person which said to make xDrip+ hardware source Nightscout follower, full guide here, it is the first comment: https://www.reddit.com/r/Bubblan/comments/roycx0/xdrip_integration_with_diabox/j20wfzv/?context=3

I get the error pointed out by delllibrary.

**Comment by nigelkeirross** [comment]
Hi

I want to migrate to Libre 2 by evolution, not revolution. I have lots of
experience with using bridges with xDrip-plus (mostly what to do with
missed readings etc)
so I think this will help with the migration. I am also going away for 2.5
weeks in January, so I was looking for a lower-risk transition for my
travels.
It is my plan to try the direct connection (bridgeless) method with OOP2
when I am more "stable" in February. That solution has lots of appeal, but
in my mind
is a larger leap, and I felt more risk than I wanted just now.

Do you happen to know if I have settings that can toggle xDrip needing OOP2
on/off?
Any thoughts on the increased xdrip cpu/battery usage? This is quite a
concern for me as battery is not great.

Many thanks
Nigel

**Comment by tzachi-dar** [comment]
I'll try to take a look later on xDrip code, but I don't think there is a setting that can help.
What probably happens is that bubble is decrypting the libre 2 data but still sending the data as libre 2.
When it reaches xDrip it is being treated as libre2, so it is sent to the oop2. There we try to break the encryption but fails, but this is a long process as it makes a lot of tries.
Then data goes back to xDrip, and now it is assumed to be OK, so checksum is checked and data is handled.
I wonder if breaking the encryption has such an effect, but this is the only thing that I can think of.
By the way, if you will try a very old bubble fw there is a chance it would send the data as encrypted and all will work well (or not, I don't know how old that would be).
I think that you can give the bridgeless a chance, most likely it will work well.

**Reply by nigelkeirross** [reply]
Thanks. The Android system shows that it is xDrip+ taking 15% of battery, but nothing for OOP2, so it would seem xDip is doing extra work. In the scenario you mention, would it be possible to do a checksum of the incoming data to see if encrypted or not, if that was cheap?
I do have an old MiaoMiao2 that I could try this on, as I believe that they do not claim to decrypt Libre 2.
Is it possible to switch from bridged to unbridged with the same sensor, ie after it has worked with a bridge? If so I can try running direct, without Bubble  this at the end of this current sensor. I assume I have to stop the sensor in xDrip+.
![xdrip-15percent](https://user-images.githubusercontent.com/21008320/208759182-57ea20a3-67ca-48be-a16c-7a68493a5838.png)

**Comment by tzachi-dar** [comment]
It is possible to switch between a bridge and a non bridge very easily.
To return to bubble, go to bluetooth scan and select the bubble device.
To use bridgeless mode, set once in setting:
Use NFC feature:               on
Starting bluetooth connection with libre 2 sensor: Always connect to libre2 sensors

and then you can scan the sensor with the phone (using nfc) and you should go to bridgelese mode.
See also: https://www.minimallooper.com/post/how-to-setup-freestyle-libre-2-and-oop2-to-use-a-native-bluetooth-connection-in-xdrip#viewer-5u227

**Reply by nigelkeirross** [reply]
OK, thanks for the instructions. I will do that toward the end of this sensor.
Two other updates: I tried to repeat using a miaomiao2 & my backup phone, as you had suggested to compare with Bubble bridge, BUT the mm2 would not even recognise the Libre 2 sensor. The event log was full of "No sensor has been found" messages, despite various "tricks" to restart that have always worked in the past. So I think my mm2 (Tomato HW 1001, FW 0004) cannot connect at all to EU Libre 2. Secondly, the very high xDrip+ battery usage seems to have disappeared overnight - now down to 1%, so I suspect the very high usage was only at the start of the new sensor. This would include about 30 mins of running xDrip+ & Bubble but without OOP2 installed on the phone. Sorry for the false alarm about this. So all good for now, with the remaining question being that the Bubble does seem to decrypt thr Libre2 data, as in the Event Log the data from the Bubble section is identical to the data from the LibreOOPAlgorithm section.
